packages/dds/cell/src/cell.ts

```diff
@@ -212,21 +201,14 @@ export class SharedCell<T = any> extends SharedObject<ISharedCellEvents<T>>
      * @returns the snapshot of the current state of the cell
      */
     protected snapshotCore(serializer: IFluidSerializer): ITree {
-        // Get a serializable form of data
-        const content: ICellValue = {
-            type: ValueType[ValueType.Plain],
-            value: this.toSerializable(this.data, serializer),
-        };
-
-        // And then construct the tree for it
         const tree: ITree = {
             entries: [
                 {
                     mode: FileMode.File,
                     path: snapshotFileName,
                     type: TreeEntry.Blob,
                     value: {
-                        contents: JSON.stringify(content),
+                        contents: this.serializer.stringify(this.data, this.handle),
```

Comment:
We don't need to stringify the contents since the runtime will invoke IFluidSerializer.stringify(..) on the returned ITree.